buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java

```diff
@@ -291,13 +296,19 @@ public static String javaVersion(JavaVersion version) {
 
       for (SubpluginOption option : options) {
         optionBuilder.add("-P");
-        optionBuilder.add(
-            "plugin:org.jetbrains.kotlin.allopen:" + option.getKey() + "=" + option.getValue());
+          optionBuilder.add(
+              "plugin:org.jetbrains.kotlin.allopen:" + option.getKey() + "=" + option.getValue());
+        }
       }
+    }
+    if (isTest) {
+      // For Kotlin tests, a special extra friend-paths argument needs to be passed to read
+      // internal elements.
+      // https://github.com/uber/okbuck/issues/709
 
-      return optionBuilder.build();
+      optionBuilder.add("-Xfriend-paths=$(location :" + JvmBuckRuleComposer.src(this) + "[output])");
```

Comment:
replace with buck-out path
```
      String composedTargetName = JvmBuckRuleComposer.src(this);
      optionBuilder.add(
          "-Xfriend-paths="
              + Paths.get("buck-out/gen")
                  .resolve(getProject().getRootProject().relativePath(getProject().getProjectDir()))
                  .resolve(String.format("lib__%s__output", composedTargetName))
                  .resolve(composedTargetName + ".jar")
                  .toString());
```
`:src_main` is already a dependency of test target which means the folder would be materialized before the tests are run.

Reply:
this would output `'-Xfriend-paths=buck-out/gen/libraries/kotlinlibrary/lib__src_main__output/src_main.jar',` for `libraries/kotlinlibrary` in the `extra_kotlinc_arguments`

Reply:
also, add a comment explaining usage buck-out path directly, which would be replace with a target once macro support is added to `extra_kotlin_arguments`

Reply:
Done